core/src/test/scala/io/snappydata/SnappyFunSuite.scala

```diff
@@ -40,23 +40,48 @@ abstract class SnappyFunSuite
   protected var testName: String = _
   protected val dirList = ArrayBuffer[String]()
 
+
   protected def sc: SparkContext = {
     val ctx = SnappyContext.globalSparkContext
-    if (ctx != null && !ctx.isStopped) ctx
-    else new SparkContext(newSparkConf())
+    if (ctx != null && !ctx.isStopped) {
+      ctx
+    } else {
+      cachedContext = null
+      new SparkContext(newSparkConf())
+    }
   }
 
   protected def sc(addOn: (SparkConf) => SparkConf): SparkContext = {
     val ctx = SnappyContext.globalSparkContext
-    if (ctx != null && !ctx.isStopped) ctx
-    else new SparkContext(newSparkConf(addOn))
+    if (ctx != null && !ctx.isStopped) {
+      ctx
+    }
+    else {
+      cachedContext = null
+      new SparkContext(newSparkConf(addOn))
+    }
   }
 
   protected def scWithConf(addOn: (SparkConf) => SparkConf): SparkContext = {
     new SparkContext(newSparkConf(addOn))
   }
 
-  protected def snc: SnappyContext = SnappyContext.getOrCreate(sc)
+  private var cachedContext : SnappyContext = null
+
+  def getOrCreate(sc: SparkContext): SnappyContext = {
+    val gnc = cachedContext
+    if (gnc != null) gnc
+    else synchronized {
+      val gnc = cachedContext
+      if (gnc != null) gnc
+      else {
+        cachedContext = SnappyContext(sc)
+        cachedContext
+      }
+    }
+  }
+
+  protected def snc: SnappyContext = getOrCreate(sc)
 
```

Comment:
Yes Sumedh, I forgot to mention that here. That becomes a bit problematic when tests stops SparkContext on their own. In that case the lazy val still points to the stale SnappyContext.